bowling/tests/bowlingGame.ut.cpp

```diff
@@ -0,0 +1,64 @@
+#include "catch.hpp"
+
+#include "bowlingGame.hpp"
+
+#include <sstream>
+
+using Catch::Matchers::StartsWith;
+
+SCENARIO("BowlingGame class can check if help is to be printed and trigger printing", "bowlingGame")
+{
+    int testArgc = 2;
+    std::string expectedHelpBeginning{"BowlingGame <input directory> <output file>"};
+
+    {
+        const char* testArgv[] = {"programName", "--help", NULL};
+
+        GIVEN("argc = " << testArgc
+                        << " and argv = {\""
+                        << (testArgv[0] ? testArgv[0] : "") << "\", \""
+                        << (testArgv[1] ? testArgv[1] : "") << "\", \""
+                        << (testArgv[2] ? testArgv[2] : "") << "\"}")
+        {
+            WHEN("BowlingGame object is created")
+            {
+                BowlingGame testBowlingGame(testArgc, testArgv);
+                THEN("isHelpToBePrinted() reeturns true and help is printed")
+                {
+                    REQUIRE(testBowlingGame.isHelpToBePrinted());
+
+                    std::stringstream ss;
+                    ss << testBowlingGame.printHelp();
+                    REQUIRE_FALSE(ss.str().empty());
+                    REQUIRE_THAT(ss.str(),
+                                 StartsWith(expectedHelpBeginning));
+                }
+            }
+        }
+    }
+    {
+        const char* testArgv[] = {"programName", "-h", NULL};
+
+        GIVEN("argc = " << testArgc
+                        << " and argv = {\""
+                        << (testArgv[0] ? testArgv[0] : "") << "\", \""
+                        << (testArgv[1] ? testArgv[1] : "") << "\", \""
+                        << (testArgv[2] ? testArgv[2] : "") << "\"}")
+        {
+            WHEN("BowlingGame object is created")
+            {
+                BowlingGame testBowlingGame(testArgc, testArgv);
+                THEN("isHelpToBePrinted() reeturns true and help is printed")
+                {
+                    REQUIRE(testBowlingGame.isHelpToBePrinted());
+
+                    std::stringstream ss;
+                    ss << testBowlingGame.printHelp();
+                    REQUIRE_FALSE(ss.str().empty());
+                    REQUIRE_THAT(ss.str(),
+                                 StartsWith(expectedHelpBeginning));
+                }
+            }
+        }
+    }
+}
```

Comment:
what about whole application tests? You should check if proper values are printed in an output file / screen. You can refactor you showScore / saveScore functions to use some indirect function that returns an output string and check it here. 
In fact this is the only major problem in your code. The rest of my comments is really a minor thing :)
You can also use mocks to provide your own content of files.

Reply:
thank You for review, i tried to fix some issues, i hope it's everything allright now